### DISCUSSION TITLE
help setting up minecraft-server-hibernation

### DISCUSSION THREAD

**Discussion by gekigek99** [gekigek99/minecraft-vanilla-server-hibernation#77] [Q&A] [answered]
from @Swaglikeobama
> hi im very new to this sorta stuff and ive managed to configure the config as to my server but what im having problems is how i run it? its probably something very obvious but i cant seem to know how :/

Hi I moved your question here so it's easier to find in the future for others...

First of all, on what OS is your minecraft server?
Can you access the minecraft server normally? (through port 25565, starting it manually)
Would you mind sending here your config file? (sometimes people forget to set something and it does not work)

**Comment by gekigek99** [comment, marked as answer]
@Swaglikeobama do you still need help?

**Reply by AndreasFlodstrom** [reply]
hi yes, sorry for slow answer ive had some big school projects i had to do, well my OS is windows 10 and yes the server can open and function totally normal, the only problem i have is that i dont really know how to start to Go script or whatever u need to do for the script to start running, im very new to this so thats why, ive configured everything and checked so its right multiple times so thats right its just i dunno how to run the script...

**Reply by gekigek99** [reply]
yea I no problem!
Go to the releases and download the latest binary (already compiled) then change the config file to what you need.
Just launch the .exe with a double click (you should then be able to connect to _yourserveraddress_:25555 ) (if the server is on the same machiene of the minecraft client just use 127.0.0.1 as _yourserveraddress_)

(remember that the config and the msh executable need to be in the same folder)

**Reply by jhendrix61978** [reply]
attempting to run getting this error 

```
v2.3.0
visit my github page: https://github.com/gekigek99
remember to give a star to this repository!
2021/03/08 17:04:40 loadConfig: invalid character 'U' in string escape code
```

`C:\Users\jhend\Desktop\forgeserver`

what did i miss

**Reply by gekigek99** [reply]
it's because the `\` character is interpreted as an escape character.

you should write `C:\\Users\\jhend\\Desktop\\forgeserver`

**Reply by jhendrix61978** [reply]
ok that solve that issue now no one can connect the server it says listening for connections all ports have been forward properly and can connect when the server is running with out script but once running the script says waiting for connection and on client says cannot connect and never see a connection in cmd window?

**Reply by gekigek99** [reply]
ok I think I understand but please be clear because it's a little confused your explanation...

are you running the minecraft client and server on the same client?
on what port is the minecraft server and on what port is the msh listening?
is the minecraft server client configured to access the server through port 25555?

**Reply by jhendrix61978** [reply]
no i have a separate server box in home server is set to port 25565 as normal i can start the forge server and connect with all pc in my home and friends can connect just fine when i turn the server on using your script it comes up in the command window waiting for connections but then when we try to connect there is no attempt shown in cmd window and client says no connection to server.

Update: changed port info in config.json and in serverpropertys.config now getting this info

```
[36m[20:31:45] [Server thread/INFO] [minecraft/DedicatedServer]: Starting Minecraft server on 192.168.1.3:25555←[0m
←[36m[20:31:45] [Server thread/INFO] [minecraft/NetworkSystem]: Using default channel type←[0m
←[36m[20:31:45] [Server thread/WARN] [minecraft/DedicatedServer]: **** FAILED TO BIND TO PORT!←[0m
←[36m[20:31:45] [Server thread/WARN] [minecraft/DedicatedServer]: The exception was: java.net.BindException: Address already in use: bind←[0m
←[36m[20:31:45] [Server thread/WARN] [minecraft/DedicatedServer]: Perhaps a server is already running on that port?←[0m
←[36m[20:31:45] [Server thread/INFO] [minecraft/MinecraftServer]: Stopping server←[0m
←[36m[20:31:45] [Server thread/INFO] [minecraft/MinecraftServer]: Saving worlds←[0m
←[36m[20:31:45] [Server thread/INFO] [FML]: The state engine was in incorrect state POSTINITIALIZATION and forced into state SERVER_STOPPED. Errors may have been discarded.←[0m
←[36m[20:31:45] [Server Shutdown Thread/INFO] [minecraft/MinecraftServer]: Stopping server←[0m
←[36m[20:31:45] [Server Shutdown Thread/INFO] [minecraft/MinecraftServer]: Saving worlds←[0m
```

there is nothing running on that port but the server so i am unclear whats causing it and ports are defenently open in router

**Reply by jhendrix61978** [reply]
ok now have what looks like full start on server but when client trys to connect we are getting this 

```
2021/03/09 21:03:38 closing connection for: 192.168.1.2
2021/03/09 21:03:46 *** jhendrix1978 tried to join from 192.168.1.2:4250 to 192.168.1.3:25565 during server startup
2021/03/09 21:03:46 closing connection for: 192.168.1.2
```

never gets past warming up stage

**Reply by jhendrix61978** [reply]
```
{
  "Basic": {
    "ServerDirPath": "C:\\Users\\jhend\\Desktop\\forgeserver",
    "ServerFileName": "forge-1.12.2-14.23.5.2854.jar",
    "StartMinecraftServer": "java -Xms16G -Xms5G -jar forge-1.12.2-14.23.5.2854.jar nogui",
    "StopMinecraftServer": "stop",
    "StopMinecraftServerForce": "",
    "HibernationInfo": "                   \u0026fserver status:\n                   \u0026b\u0026lHIBERNATING",
    "StartingInfo": "                   \u0026fserver status:\n                    \u00266\u0026lWARMING UP",
    "TimeBeforeStoppingEmptyServer": 60,
    "CheckForUpdates": true
  },
  "Advanced": {
    "ListenHost": "192.168.1.3",
    "ListenPort": "4250",
    "TargetHost": "192.168.1.3",
    "TargetPort": "25565",
    "Debug": true,
    "ServerVersion": "1.12.2",
    "ServerProtocol": "340"
  }
}
```

**Reply by gekigek99** [reply]
- that `**** FAILED TO BIND TO PORT!` might mean that the minecraft server is already running so try to stop it or reboot the server machine
- (remember to set `TargetPort` to the port the minecraft server is running on (specified in `server.properties`))

_try doing the above 2 instructions and if it does not work keep reading..._

> never gets past warming up stage

I know that on version 2.3.0 there is a similar bug that was reported just 2 days ago

- try downloading [msh-v2.3.1-windows-x86.exe.zip](https://github.com/gekigek99/minecraft-server-hibernation/files/6115128/msh-v2.3.1-windows-x86.exe.zip) and tell me if it works

_if this does not solve the problem keep reading..._

are you running msh on your windows machine and the minecraft server on a linux server in your lan?

because that
```
"ListenHost": "192.168.1.3",
"ListenPort": "4250",
"TargetHost": "192.168.1.3",
"TargetPort": "25565",
```
is kinda strange

I would advise to run msh directly in the server machine with this config (obvously you should download the correct executable depending on your OS... tell me if you don't know how to compile it yourself)
```
"ListenHost": "0.0.0.0",
"ListenPort": "4250",
"TargetHost": "127.0.0.1",
"TargetPort": "25565",
```
(the minecraft server port in `server.properties` should be set as `25565` an you should connect to the server (within minecraft client) with `serverip:4250`)

**Reply by jhendrix61978** [reply]
so yes i have client pc running just the game and a windows server running a forge minecraft server on the same lan and server runs fine we play on it from internal lan and connections from wan are able to play we are just looking for a script to autostart and stop when noone is on
_________________________________________________________________________
UPDATE: Looks like new EXE solved the issues as I am now able to start server and connect and server shuts down after 60 seconds when all are off!

Thank you Very Much

**Reply by gekigek99** [reply]
@jhendrix61978 you are welcome! If you like this program please give a star to the rep!
